src/libs/metadata_view.c

```diff
@@ -791,46 +921,319 @@ void connect_key_accels(dt_lib_module_t *self)
-
-    d->metadata[k] = GTK_LABEL(gtk_label_new("-"));
-    gtk_widget_set_name(GTK_WIDGET(d->metadata[k]), "brightbg");
-    gtk_label_set_selectable(d->metadata[k], TRUE);
-    gtk_widget_set_halign(GTK_WIDGET(d->metadata[k]), GTK_ALIGN_FILL);
-    gtk_label_set_xalign (d->metadata[k], 0.0f);
-    gtk_label_set_ellipsize(GTK_LABEL(d->metadata[k]),
-                            k == md_exif_model || k == md_exif_lens || k == md_exif_maker
-                            ? PANGO_ELLIPSIZE_END : PANGO_ELLIPSIZE_MIDDLE);
-    if(k == md_internal_filmroll)
+    GtkTreeIter sort_iter;
+    gboolean valid = gtk_tree_model_get_iter_first(d->sort_model, &sort_iter);
+    while(valid)
     {
-      // film roll jump to:
-      g_signal_connect(G_OBJECT(GTK_WIDGET(d->metadata[k])), "button-press-event", G_CALLBACK(_filmroll_clicked), NULL);
+      GtkTreeIter iter;
+      int index;
+      gboolean visible;
+      char *name = NULL;
+      gtk_tree_model_get(d->sort_model, &sort_iter,
+                         DT_METADATA_COL_INDEX, &index,
+                         DT_METADATA_COL_NAME_L, &name,
+                         DT_METADATA_COL_VISIBLE, &visible,
+                         -1);
+      if(!_is_metadata_ui(index))
+        continue;
+      gtk_list_store_append(store, &iter);
+      gtk_list_store_set(store, &iter,
+                         DT_METADATA_PREF_COL_INDEX, index,
+                         DT_METADATA_PREF_COL_NAME_L, name,
+                         DT_METADATA_PREF_COL_VISIBLE, visible,
+                         -1);
+      g_free(name);
+      valid = gtk_tree_model_iter_next(d->sort_model, &sort_iter);
     }
+  }
+
+  GtkWidget *view = gtk_tree_view_new_with_model(model);
+  g_object_unref(model);
+  GtkCellRenderer *renderer = gtk_cell_renderer_text_new();
+  GtkTreeViewColumn *column = gtk_tree_view_column_new_with_attributes(_("metadata"), renderer,
+                                                    "text", DT_METADATA_PREF_COL_NAME_L, NULL);
+  gtk_tree_view_column_set_expand(column, TRUE);
+  gtk_tree_view_append_column(GTK_TREE_VIEW(view), column);
+  renderer = gtk_cell_renderer_toggle_new();
+  g_signal_connect(renderer, "toggled", G_CALLBACK(_select_toggled_callback), store);
+  column = gtk_tree_view_column_new_with_attributes(_("visible"), renderer,
+                                                    "active", DT_METADATA_PREF_COL_VISIBLE, NULL);
+  gtk_tree_view_append_column(GTK_TREE_VIEW(view), column);
+
+  // drag & drop
+  gtk_tree_view_set_reorderable(GTK_TREE_VIEW(view), TRUE);
+  g_signal_connect(G_OBJECT(model), "row-inserted", G_CALLBACK(_drag_data_inserted), NULL);
+
+  // workaround to avoid verticall scrollbar to cover view content
+  GtkWidget *grid = gtk_grid_new();
+  gtk_grid_attach(GTK_GRID(grid), view, 0, 0, 1, 1);
+  GtkWidget *scroll = gtk_label_new("  ");
+  gtk_grid_attach(GTK_GRID(grid), scroll, 1, 0, 1, 1);
+  gtk_container_add(GTK_CONTAINER(w), grid);
```

Comment:
This isn't pretty and when you increase the size of the window vertically, the first column no longer expands if you also increase the horizontal size. I know it doesn't need to, because it has allocated enough space, but it is still somewhat ugly and not standard behavior.

I would suggest above using 
```
  gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(w), GTK_POLICY_NEVER, GTK_POLICY_ALWAYS);
  gtk_scrolled_window_set_overlay_scrolling(GTK_SCROLLED_WINDOW(w), FALSE);
```
and here simply
`  gtk_container_add(GTK_CONTAINER(w), view);
`
It is not _perfect_, because the scrollbar stays, even if it isn't needed. But most of the time it is needed anyway...

Reply:
That's better indeed. I take it!
Thanks for your help (as usual :smiley: )